pkg/vmprovider/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -168,9 +167,7 @@ func vmTests() {
 					// Update the VM Class with the XML.
 					vmClass := &vmopv1alpha1.VirtualMachineClass{}
 					Expect(ctx.Client.Get(ctx, client.ObjectKey{Name: vm.Spec.ClassName}, vmClass)).To(Succeed())
-					vmClass.Spec.ConfigSpec = &runtime.RawExtension{
-						Raw: w.Bytes(),
-					}
+					vmClass.Spec.ConfigSpec = w.Bytes()
```

Comment:
Would `json.RawMessage(obj_data)` be more readable and explicit here?

Reply:
The type conversion is implicit so this is unecessary.

Reply:
> If `type T K` then you can `var t T; var k K; t = k`.

The above assertion doesn't not work for non-slice primitive types. Will dig into this more, but ([Go playground](https://go.dev/play/p/N98cmmW5Tuq):

```go
package main

import (
	"fmt"
)

type Int int

func main() {
	var I Int = 1
	var i int = I
	fmt.Println(i)
}
```

does not work, and will emit:

```
./prog.go:11:14: cannot use I (variable of type Int) as type int in variable declaration

Go build failed.
```

While ([Go playground](https://go.dev/play/p/LsTxA33zd6k)) _does_ work:

```go
package main

import (
	"fmt"
)

type SliceOfInts []int

func main() {
	var ints []int
	var sois SliceOfInts = SliceOfInts{1, 2, 3}
	ints = sois
	fmt.Println(ints)
}
```

and emits:

```
[1 2 3]

Program exited.
```